onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -162,7 +162,18 @@ void BasicBackend::EnableGPUThrottling(ov::AnyMap& device_config) {
 }
 
 void BasicBackend::EnableStreams() {
-  global_context_.ie_core.SetStreams(global_context_.device_type, global_context_.num_streams);
+  // Streams can be set only if the device is not one of AUTO, MULTI, or HETERO
+  // Throw an exception if the user tries to set num_streams for these devices
```

Comment:
Can we have this check during session options creation itself. openvino_provider_factory should be the right place as it validates all the command line options.